proteuslib/unit_models/nanofiltration_0D.py

```diff
@@ -197,46 +303,210 @@ def build(self):
+                ub = 3e-2
+                lb = 1e-4
+            elif j in solute_set:
+                ub = 1e-3
+                lb = 1e-8
+            return lb, ub
+
+        self.flux_mass_io_phase_comp = Var(
             self.flowsheet().config.time,
+            self.io_list,
             self.config.property_package.phase_list,
             self.config.property_package.component_list,
-            initialize=1e-3,
-            bounds=(1e-12, 1e6),
+            initialize=flux_mass_io_phase_comp_initialize,
+            bounds=flux_mass_io_phase_comp_bounds,
             units=units_meta('mass')*units_meta('length')**-2*units_meta('time')**-1,
-            doc='Flux at feed inlet')
-        self.flux_mass_phase_comp_out = Var(
+            doc='Mass flux across membrane at inlet and outlet')
+
+        self.avg_conc_mass_io_phase_comp = Var(
+            self.flowsheet().config.time,
+            self.io_list,
+            self.config.property_package.phase_list,
+            solute_set,
+            initialize=10,
+            bounds=(1e-3, 2e3),
+            units=pyunits.kg * pyunits.m ** -3,
+            doc="Average mass concentration at inlet and outlet")
+
+        self.area = Var(
+            initialize=10,
+            bounds=(1e-1, 1e3),
+            domain=NonNegativeReals,
+            units=units_meta('length') ** 2,
+            doc='Membrane area')
+
+        def recovery_mass_phase_comp_initialize(b, t, p, j):
+            if j in solvent_set:
+                return 0.1
+            elif j in solute_set:
+                return 0.01
+
+        def recovery_mass_phase_comp_bounds(b, t, p, j):
+            ub = 1 - 1e-6
+            if j in solvent_set:
+                lb = 1e-2
+            elif j in solute_set:
+                lb = 1e-5
+            return lb, ub
+
+        self.recovery_mass_phase_comp = Var(
             self.flowsheet().config.time,
             self.config.property_package.phase_list,
             self.config.property_package.component_list,
-            initialize=1e-3,
-            bounds=(1e-12, 1e6),
-            units=units_meta('mass')*units_meta('length')**-2*units_meta('time')**-1,
-            doc='Flux at feed outlet')
-        self.avg_conc_mass_phase_comp_in = Var(
+            initialize=recovery_mass_phase_comp_initialize,
+            bounds=recovery_mass_phase_comp_bounds,
+            units=pyunits.dimensionless,
+            doc='Mass-based component recovery')
+
+        self.recovery_vol_phase = Var(
             self.flowsheet().config.time,
             self.config.property_package.phase_list,
-            self.solute_list,
-            initialize=1e-3,
-            bounds=(1e-8, 1e6),
-            domain=NonNegativeReals,
-            units=units_meta('mass')*units_meta('length')**-3,
-            doc='Average solute concentration at feed inlet')
-        self.avg_conc_mass_phase_comp_out = Var(
+            initialize=0.1,
+            bounds=(1e-2, 1 - 1e-6),
```

Comment:
Why is the upper bound 1 - 1e-6? Is there some significance to it? Can it not just be 1?

Reply:
This was taken from previous code. I think one reason is because in certain places, there could be implementation of `1- self.recovery_vol_phase`. That could lead to division by zero.

Also, you can't realistically have a recovery rate of 1, so this is just meant to set an upper limit that approaches 1 (even that is not really practical in reality though).

Reply:
In these cases I would personally use 1 for two reasons.

1. It represents the limit of recovery/removal. which is one interpretation of a bound.
2. Doing `1-eps` applies an arbitrary limit on the maximum removal/recovery. It probably does not matter in practice, but it just feels wrong to me.

On the matter of the potential division by zero, that would be better addressed by trying to formulate the constraint such that a division by zero cannot occur. Even with an arbitrary bound on the approach to 0, you still need to deal with the non-linearities of approaching a singularity.

Reply:
I don't see any other reason not to change this upper bound to 1 as long as it doesn't cause any unexpected issues. The division by zero issue was just my guess. @TimBartholomew do you recall the significance of setting this upper bound to `1 - 1e-6`?